### PR TITLE
OpcodeDispatcher: Optimizes scalar movd/movq

### PR DESCRIPTION
**Pull request by Sonicadvance1**
MMX and SSE versions are now optimal.

### CODE REVIEW

**Review by lioncash — COMMENTED** [review]
One minor nit, but looks good overall

**Comment by Sonicadvance1** [comment]
> One minor nit, but looks good overall

Nit wasn't posted?

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/Vector.cpp [hunk 1]
```diff
@@ -2256,20 +2256,33 @@ void OpDispatchBuilder::VPMASKMOVOp<true>(OpcodeArgs);
...
+    else {
+      // Loading from Memory as a scalar. Zero extend
+      OrderedNode *Src = LoadSource(FPRClass, Op, Op->Src[0], Op->Flags, -1);
+      StoreResult(FPRClass, Op, Op->Dest, Src, -1);
+    }
   }
   else {
-    // Destination is GPR or mem
-    // Extract from XMM first
     auto ElementSize = GetDstSize(Op);
```
Comment (lioncash) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher/Vector.cpp:2273]: Looks like this can be moved into the GPR case below

**Comment by lioncash** [comment]
Weird, had to repost it

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/Vector.cpp (hunk 1 above)
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/Vector.cpp]: Moved.

**Review by lioncash — APPROVED** [review]

**Pull request merged**